src/__tests__/functional-tests/test-utils.test.tsx

```diff
@@ -187,40 +187,46 @@ describe.each(components)('ElementWrapper selectors for %s component', component
   });
 });
 
-describe.each(componentsWithNoExtraFinders)('ElementWrapper selectors for %s component', componentName => {
+describe.only.each(componentswithseveralroots)('ElementWrapper selectors for %s component', componentName => {
```

Comment:
These two components `<AppLayout />` and `<TopNavigation />` will still have some limited functionality for the same reason [mentioned here](https://github.com/cloudscape-design/components/pull/2932/files#r1831055671).

These two cases will not work:

1. Adding custom selector in unit tests:
```tsx
findAllAppLayouts('[data-my-custom-attrib]')

// Generated selector:
// .awsui_root_hash[data-my-custom-attrib]

<div data-my-custom-attrib>   // <--- data attribute is added here
  <div class="awsui_root_hash"> // <--- actual root class is here
    ...
  </div>
</div>
```

2. Using `get()` in integration tests:
```tsx
findAllAppLayouts().get(2)

// Generated selector
// .awsui_root_hash:nth-child(2);

<div>
  <div class="awsui_root_hash">...</div>
</div>
<div>
  <div class="awsui_root_hash">...</div>
</div>
```